src/main/java/io/vertx/httpproxy/interceptors/impl/HeadersFilterInterceptorImpl.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2011-2024 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.httpproxy.interceptors.impl;
+
+import java.util.Set;
+
+public class HeadersFilterInterceptorImpl extends HeadersInterceptorImpl {
```

Comment:
Type inheritance isn't necessary here. You could have a single class with two constructors.

Type inheritance has its merits but it has several drawbacks in terms of object design and performance.